app/controllers/subscriptions_controller.rb

```diff
@@ -0,0 +1,24 @@
+require_relative "../models/mailchimp"
+
+class SubscriptionsController < ApplicationController
+  # Subscribe to the MailChimp newsletter.
+  def create
+    redirect_to request.referer if !current_user
```

Comment:
Check out `rubocop` locally; the CI didn't display it in the output because of the test coverage failure (if it'd be more helpful I can make rubocop either run first or ensure both always run?).